hack/validate-manifests/helm.go

```diff
@@ -0,0 +1,166 @@
+package main
+
+import (
+	"fmt"
+	"path/filepath"
+
+	fluxhelmv2beta1 "github.com/fluxcd/helm-controller/api/v2beta1"
+	"github.com/fluxcd/pkg/runtime/transform"
+	fluxsourcesv1beta1 "github.com/fluxcd/source-controller/api/v1beta1"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/chartutil"
+	"helm.sh/helm/v3/pkg/cli"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+type HelmReleaseValidator struct {
+	helmRelease *fluxhelmv2beta1.HelmRelease
+}
+
+func (v *HelmReleaseValidator) Name() string {
+	return "HelmRelease " + v.helmRelease.Name
+}
+
+func (v *HelmReleaseValidator) Check(ctx *Context) (done bool, errs []error) {
+	defer func() {
+		if done {
+			ref := metasToRef(v.helmRelease.TypeMeta, v.helmRelease.ObjectMeta)
+			ctx.MarkChecked(ref)
+		}
+	}()
+
+	for _, dep := range v.helmRelease.Spec.DependsOn {
+		ref := NewObjectRef(
+			v.helmRelease.APIVersion, v.helmRelease.Kind,
+			dep.Namespace, dep.Name,
+		)
+		if ref.Metadata.Namespace == "" {
+			ref.Metadata.Namespace = v.helmRelease.Namespace
+		}
+		if !ctx.IsChecked(ref) {
+			return false, nil
+		}
+	}
+
+	ctx.V(1).Infof("Validating Helm release: %q", v.helmRelease.Name)
+	rendered, err := renderHelmRelease(ctx, v.helmRelease)
+	if err != nil {
+		return true, []error{err}
+	}
+	return true, validateResource(ctx, rendered)
+}
+
+func addHelmRepo(ctx *Context, helmRepo *fluxsourcesv1beta1.HelmRepository) {
+	ctx.V(1).Infof("Adding Helm repository: %s", helmRepo.Name)
+	ctx.SetObject(metasToRef(metav1.TypeMeta{Kind: helmRepo.Kind}, helmRepo.ObjectMeta), helmRepo)
+}
+
+func renderHelmRelease(ctx *Context, helmRelease *fluxhelmv2beta1.HelmRelease) ([]byte, error) {
+	actionConfig := new(action.Configuration)
+	err := actionConfig.Init(genericclioptions.NewConfigFlags(false), "", "memory", func(string, ...interface{}) {})
+	if err != nil {
+		return nil, err
+	}
+	client := action.NewInstall(actionConfig)
+	client.DryRun = true
+	client.ReleaseName = helmRelease.Spec.ReleaseName
+	if client.ReleaseName == "" {
+		client.ReleaseName = helmRelease.Name
+	}
+	client.Replace = true // Skip the name check
+	client.ClientOnly = true
+	client.IncludeCRDs = true
```

Comment:
Should this come from the `HelmRelease` configuration?

```
	// SkipCRDs tells the Helm install action to not install any CRDs. By default,
	// CRDs are installed if not already present.
	//
	// Deprecated use CRD policy (`crds`) attribute with value `Skip` instead.
	//
	// +deprecated
	// +optional
	SkipCRDs bool `json:"skipCRDs,omitempty"`
```

Reply:
It's not used by any application currently. Do you think we would ever need to install a Helm chart without its CRDs?